numpy/random/_generator.pyx

```diff
@@ -559,39 +558,39 @@ cdef class Generator:
             high = low
```

Comment:
Documentation at https://github.com/numpy/numpy/pull/15511/files#diff-f166b929b9f07d28bd1582546a65c87bR500 should be updated, perhaps something like:
```diff
- dtype : {str, dtype}, optional
+ dtype : dtype, optional
-           Desired dtype of the result. All dtypes are determined by their
-           name, i.e., 'int64', 'int', etc, so byteorder is not available
-           and a specific precision may have different C types depending
+           Desired dtype of the result. Byteorder must be native, and the result
+           dtype will match the size but perhaps not the underlying C type, depending
            on the platform. The default value is `np.int_`.
```

Reply:
This comment still applies, there are multiple pieces of documentation like this

Reply:
You are right, I updated it. 
What about repeating part: "_All dtypes are determined by their name_". All occurrences of _dtype.name_ were replaced... it should be rather: "All dtypes are determined by their types", right?

Reply:
I don't think there's any replacement for that sentence. I tried to capture the behavior with _"the result dtype will be equivalent but may not be identical"_. To elaborate on that, what I mean is that we guarantee
`func(dtype=dt).dtype == dt`, but not that `func(dtype=dt).dtype is dt`. I don't know if we want to mention that explicitly in the docs

Really, what we're trying to cover here is (on linux) someone passing `np.longlong`, and being surprised when they get back `np.int_`, even though both are 64-bit.

Maybe that's not interesting and we should just remove the remark entirely. Perhaps another maintainer can comment on whether we need it.

Reply:
it is a bit confusing. I think we can remove ", and the result dtype will be equivalent but may not be identical", no need to dive into the weeds here.

Reply:
Ok, updated.